### DISCUSSION TITLE
BC Universal Code: External Tables

### DISCUSSION THREAD

**Discussion by Davide-Mendola-EOS** [microsoft/ALAppExtensions#20622] [Q&A] [answered]
Hi,
i need help about this topic: External Tables are under "Universal Code" Microsoft's rules?

Exatly, if i use an external table for developing on my OnPrem customer, MS sign it as "OUT-Universal Code"?

i need this answer to bring decision about an project.

Thank you for your support!

Have a nice day.

**Comment by mazhelez** [comment]
@JesperSchulz do you know the answer to that?

**Comment by SBalslev** [comment, marked as answer]
Currently only some table types are universal code compatible (CDS/CRM). You can verify by creating an extension with "target"="cloud" in the app.json and a external table type definition for your needs.
